src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php

```diff
@@ -61,24 +61,27 @@ public function testDefinitions()
 
     public function testCreateDeprecatedService()
     {
-        $definition = new Definition('stdClass');
-        $definition->setDeprecated(true);
+        $deprecations = array();
+        set_error_handler(function ($type, $msg) use (&$deprecations) {
+            if (E_USER_DEPRECATED !== $type) {
+                restore_error_handler();
 
-        $that = $this;
-        $wasTriggered = false;
+                throw new \LogicException(sprintf('Unexpected error: "%s".', $msg));
```

Comment:
what about forwarding to phpunit's error handler instead?
`return call_user_func_array('PHPUnit_Util_ErrorHandler::handleError', func_get_args());`